python/cudf/cudf/core/index.py

```diff
@@ -428,8 +428,51 @@ def __array_function__(self, func, types, args, kwargs):
         else:
             return NotImplemented
 
-    def isin(self, values):
-        return self.to_series().isin(values)
+    def isin(self, values, level=None):
+        """Return a boolean array where the index values are in values.
+
+        Compute boolean array of whether each index value is found in
+        the passed set of values. The length of the returned boolean
+        array matches the length of the index.
+
+        Parameters
+        ----------
+        values : set or list-like
```

Comment:
Can we allow this to be an Index / MultiIndex as well for more efficient paths?

Reply:
Added `Index` support for `values`